test/functional/api/buffer_spec.lua

```diff
@@ -435,6 +437,96 @@ describe('api/buf', function()
     end)
   end)
 
+  describe('nvim_buf_set_keymap', function()
```

Comment:
if it's not too much trouble could you move this to `api/keymap_spec.lua`? That's where the tests for  `nvim_[buf_]_get_keymap` live.